vm/systemSmartContracts/staking.go

```diff
@@ -1048,6 +1061,19 @@ func (r *stakingSC) switchJailedWithWaiting(args *vmcommon.ContractCallInput) vm
 	return vmcommon.Ok
 }
 
+func (r *stakingSC) isActionAllowed(registrationData *StakedData, blsKey []byte) bool {
+	if registrationData.Jailed {
```

Comment:
You can replace all with one liner:
return registrationData.Jailed || r.eei.CanUnJail(blsKey) || r.eei.IsBadRating(blsKey)

or

isNodeJailedOrWIthBadRating := registrationData.Jailed || r.eei.CanUnJail(blsKey) || r.eei.IsBadRating(blsKey
return isNodeJailedOrWIthBadRating

And then you should change in the caller side the condition from !r.isNodeJailedOrWIthBadRating to r.isNodeJailedOrWIthBadRating

Reply:
Done